test/framework/filetools.py

```diff
@@ -355,6 +355,23 @@ def test_is_readable(self):
         os.chmod(test_file, 0)
         self.assertFalse(ft.is_readable(test_file))
 
+    def test_symlink_readlink_file(self):
+        """Test read link target file"""
+
+        # write_file and read_file tests are elsewhere. so not getting their states
+        fp = os.path.join(self.test_prefix, 'test.txt')
+        txt = "test_my_link_file"
+        ft.write_file(fp, txt)
+
+        link = os.path.join(self.test_prefix, 'test.link')
+        # creating the link file
+        ft.symlink(fp, link)
+
+        # reading file contents and comparing to link
+        self.assertEqual(ft.read_file(fp), ft.read_file(link))
```

Comment:
this doesn't mean at all that the files are correctly symlinked? could be a copy? just test below is enough imho, although maybe also check that `link` *is* a symbolic link (as opposed to a copied file)